cmd/server/main.go

```
@@ -123,12 +131,33 @@
 	// Initialize repositories and services
 	repos := repositories.NewRepository(store)
 	service := server.NewMetricsService(repos, app.logger)
-	serverHandlers := handlers.NewHandler(service, app.logger, app.db, app.cfg.Envs.Key, app.cfg.PrivateKey)
+	serverHandlers := handlers.NewHandler(service, app.logger, app.db, app.cfg.Envs.Key,
+		app.cfg.PrivateKey, app.cfg.TrustedSubnet)
 
+	grpcLis, err := net.Listen("tcp", ":50051")
+	if err != nil {
+		log.Fatalf("Failed to listen: %v", err)
```

golangci-lint (1.62.2)

[error] exitAfterDefer: log.Fatalf will exit, and `defer func(){...}(...)` will not run

(gocritic)